demo_nodes_cpp/src/topics/allocator_tutorial.cpp

```diff
@@ -92,6 +92,33 @@ constexpr bool operator!=(
   return false;
 }
 
+// You need to overload get_rcl_allocator for custom allocators.
+// This function needs to build an instance of rcl_allocator_t
+// for use in C code.
+template<typename T>
+rcl_allocator_t get_rcl_allocator(MyAllocator<T>)
```

Comment:
Doesn't this need to be in the `rclcpp` namespace?

Reply:
Argument-dependent lookup saves us: https://en.cppreference.com/w/cpp/language/adl. As long as the function is in the namespace of one of its arguments, the overload will be resolved.

Reply:
Yes, but should it be in the same namespace... I feel like the answer is yes, but I'm open to suggestion. I'm thinking from the perspective of a person reading this code and not knowing what it is exactly that is being overridden or where it comes from. The namespace gives important context imo.

Reply:
Got you. I don't care one way or another, so if you stumble over the namespace, it's probably a good signal that many devs would stumble over the namespace. Moved it to rclcpp.

rclcpp::allocator wouldn't work because the calling code is in rclcpp.

Reply:
It should be in the `rclcpp::allocator` namespace, I commented on the original pr: https://github.com/ros2/rclcpp/pull/1324#pullrequestreview-517938632